docs/site/express-with-lb4-rest-tutorial.md

```diff
@@ -0,0 +1,304 @@
+
+Create a new file **src/server.ts** to create your Express class:
+
+```sh
+npm install --save express
+npm install --save-dev @types/express
+```
+
+{% include code-caption.html content="src/server.ts" %}
+
+```ts
+import * as express from 'express';
+
+export class ExpressApplication {
+  constructor() {}
+}
+```
+
+Create two properties, the Express application instance and LoopBack application
+instance:
+
+```ts
+import {NoteApplication} from './application';
+import {ApplicationConfig} from '@loopback/core';
+import * as express from 'express';
+
+export class ExpressApplication {
+  private app: express.Application;
+  private lbApp: NoteApplication;
+
+  constructor(options: ApplicationConfig = {}) {
+    this.app = express();
+    this.lbApp = new NoteApplication(options);
+  }
+}
+```
+
+Now, inside the constructor, we're going to add the basepath and expose the
+front-end assets via Express:
+
+```ts
+this.app.use('/api', this.lbApp.requestHandler);
+```
+
+Let's also modify **public/index.html** to update the base path:
+
+{% include code-caption.html content="public/index.html" %}
+
+```html
+<h3>OpenAPI spec: <a href="/api/openapi.json">/openapi.json</a></h3>
+<h3>API Explorer: <a href="/api/explorer">/explorer</a></h3>
+```
+
+Then, we can add some custom Express routes, as follows:
+
+```ts
+import {Request, Response} from 'express';
+import * as path from 'path';
+
+export class ExpressApplication {
+  private app: express.Application;
+  private lbApp: NoteApplication;
+
+  constructor(options: ApplicationConfig = {}) {
+    // earlier code
+
+    // Custom Express routes
+    this.app.get('/', function(_req: Request, res: Response) {
+      res.sendFile(path.resolve('public/express.html'));
+    });
+    this.app.get('/hello', function(_req: Request, res: Response) {
+      res.send('Hello world!');
+    });
+  }
+}
+```
+
+And add the
+[public/express.html](https://github.com/strongloop/loopback-next/blob/master/examples/express-composition/public/express.html)
+file to your project.
+
+Finally, we can add functions to boot the `Note` application and start the
+Express application:
+
+```sh
+npm install --save p-event
+```
+
+```ts
+import * as pEvent from 'p-event';
+
+export class ExpressApplication {
+  private app: express.Application;
+  private lbApp: NoteApplication;
+
+  constructor(options: ApplicationConfig = {}) {
+    //...
+  }
+
+  async boot() {
+    await this.lbApp.boot();
+  }
+
+  async start() {
+    const server = this.app.listen(3000);
+    await pEvent(server, 'listening');
+  }
+}
+```
+
+Now that our **src/server.ts** file is ready, then we can modify our
+**src/index.ts** file to start the application:
+
+{% include code-caption.html content="src/index.ts" %}
+
+```ts
+import {ExpressServer} from './server';
+import {ApplicationConfig} from '@loopback/core';
+
+export {ExpressServer};
+
+export async function main(options: ApplicationConfig = {}) {
+  const server = new ExpressServer(options);
```

Comment:
ditto `ExpressApplication`. Might want to rename the assigned `server` variable too.